src/component/solenoid.go

```diff
@@ -1,26 +1,103 @@
 package component
 
+import (
+	"firecontroller/io"
+	"firecontroller/utilities"
+	"log"
+	"strconv"
+	"time"
+)
+
 // Solenoid - base component + solenoid specific metadata
 type Solenoid struct {
-	BaseComponent
-	Voltage int
-	Type    SolenoidType
+	BaseComponent `yaml:",inline"`
+	Voltage       int           `yaml:"voltage"`
```

Comment:
As discussed in person. Lets move all of the non functional descriptive data into a metadata object. This includes Voltage as that is more of a nice to have thing on the frontend side. NON-BLOCKING

Reply:
Roger roger